operator/v1/stable.ingresscontroller.testsuite.yaml

```diff
@@ -13,3 +13,406 @@ tests:
+      metadata:
+        name: default-not-allowed-values
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:          
+            request:
+            - name: X-SSL-Client-Cert 
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,hello]"
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy format string as defined in https://cbonte.github.io/haproxy-dconv/2.4/configuration.html#8.2.4 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2. Sample fetchers allowed are req.hdr, ssl_c_der. Converters allowed are lower, base64."
+  - name: Should not allow to set dynamic request header values with not allowed sample fetchers.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: default-not-allowed-values
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:          
+            request:
+            - name: X-SSL-Client-Cert 
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der1234,base64]"
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy format string as defined in https://cbonte.github.io/haproxy-dconv/2.4/configuration.html#8.2.4 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2. Sample fetchers allowed are req.hdr, ssl_c_der. Converters allowed are lower, base64."
+  - name: Should be required to specify the set field when the discriminant type is Set.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      spec:
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+    expectedError: "set is required when type is Set, and forbidden otherwise"
+  - name: Should be able to add set field only when discriminant type is Set.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      spec:
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                set:
+                  value: DENY
+    expectedError: 'Invalid value: "null"'
```

Comment:
This is not the right error, you should get a `field.Required` error as part of a compound error, please specify the test output for the error part that you expect the user to act upon

Reply:
@JoelSpeed this is what I get when the expected error is set to ''
`• [FAILED] [2.236 seconds]
 [route.openshift.io/v1, Resource=routes] [Stable] Route On Create [It] Should be required to specify the set field when the discriminant type is Set.
/home/miheer/.gvm/pkgsets/go1.20.4/global/src/github.com/openshift/api/tests/generator.go:202

  [FAILED] Unexpected error:
      <*errors.StatusError | 0xc001782aa0>: {
          ErrStatus: {
              TypeMeta: {Kind: "Status", APIVersion: "v1"},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "Route.route.openshift.io \"hello-openshift-actions\" is invalid: [spec.httpHeaders.actions.response[0].action.type: Required value, <nil>: Invalid value: \"null\": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]",
              Reason: "Invalid",
              Details: {
                  Name: "hello-openshift-actions",
                  Group: "route.openshift.io",
                  Kind: "Route",
                  UID: "",
                  Causes: [
                      {
                          Type: "FieldValueRequired",
                          Message: "Required value",
                          Field: "spec.httpHeaders.actions.response[0].action.type",
                      },
                      {
                          Type: "FieldValueInvalid",
                          Message: "Invalid value: \"null\": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation",
                          Field: "<nil>",
                      },
                  ],
                  RetryAfterSeconds: 0,
              },
              Code: 422,
          },
      }
      Route.route.openshift.io "hello-openshift-actions" is invalid: [spec.httpHeaders.actions.response[0].action.type: Required value, <nil>: Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]
  occurred
  In [It] at: /home/miheer/.gvm/pkgsets/go1.20.4/global/src/github.com/openshift/api/tests/generator.go:180 @ 07/25/23 22:02:47.714
------------------------------
SSS
------------------------------
`

Reply:
@JoelSpeed am I missing anything related to CEL condition ?

Reply:
` [FAILED] [2.236 seconds]
 [route.openshift.io/v1, Resource=routes] [Stable] Route On Create [It] Should be required to specify the set field when the discriminant type is Set.
/home/miheer/.gvm/pkgsets/go1.20.4/global/src/github.com/openshift/api/tests/generator.go:202

  [FAILED] Unexpected error:
      <*errors.StatusError | 0xc001782aa0>: {
          ErrStatus: {
              TypeMeta: {Kind: "Status", APIVersion: "v1"},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "Route.route.openshift.io \"hello-openshift-actions\" is invalid: [spec.httpHeaders.actions.response[0].action.type: Required value, <nil>: Invalid value: \"null\": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]",
              Reason: "Invalid",
              Details: {
                  Name: "hello-openshift-actions",
                  Group: "route.openshift.io",
                  Kind: "Route",
                  UID: "",
                  Causes: [
                      {
                          Type: "FieldValueRequired",
                          Message: "Required value",
                          Field: "spec.httpHeaders.actions.response[0].action.type",
                      },
                      {
                          Type: "FieldValueInvalid",
                          Message: "Invalid value: \"null\": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation",
                          Field: "<nil>",
                      },
                  ],
                  RetryAfterSeconds: 0,
              },
              Code: 422,
          },
      }
      Route.route.openshift.io "hello-openshift-actions" is invalid: [spec.httpHeaders.actions.response[0].action.type: Required value, <nil>: Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]
  occurred
  In [It] at: /home/miheer/.gvm/pkgsets/go1.20.4/global/src/github.com/openshift/api/tests/generator.go:180 @ 07/25/23 22:02:47.714`

Reply:
This is what I get when I set the expectedError: ''

Reply:
i thnk i got your point I will add Route.route.openshift.io "hello-openshift-actions" is invalid: [spec.httpHeaders.actions.response[0].action.type: Required value, : Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation] in the expected error

Reply:
In this case, I'd say the point the test cares about is `spec.httpHeaders.actions.response[0].action.type: Required value`, since that's the validation we are looking to test